packages/mpesa/src/Adaptor.js

```diff
@@ -0,0 +1,333 @@
+ * @property {object} errors - Map of errorCodes -> error messages, ie, `{ 404: 'Resource not found;' }`. Pass `false` to suppress errors for this code.
+ * @property {object} form - Pass a JSON object to be serialised into a multipart HTML form (as FormData) in the body.
+ * @property {object} query - An object of query parameters to be encoded into the URL.
+ * @property {object} headers - An object of headers to append to the request.
+ * @property {string} parseAs - Parse the response body as json, text or stream. By default will use the response headers.
+ * @property {number} timeout - Request timeout in ms. Default: 300 seconds.
+ * @property {object} tls - TLS/SSL authentication options. See https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions
+ */
+
+
+/**
+ * STKPush request data object
+ * @typedef {Object} STKPushObject
+ * @public
+ * @property {number} Amount - Amount charged.
+ * @property {number} PartyA - The phone number sending money. Expected to be a valid Safaricom Number that is M-pesa registered in the format 2547XXXXXXXX
+ * @property {number} PartyB - The organization that recerives the funds
+ * @property {number} PhoneNumber - The mobile number to receive the STK pin prompt
+ * @property {URL} CallBackURL - A valid secure URL that is used to receive notifications from M-Pesa API. It is the endpoint to which the results will be snt by M-Pesa API
+ * @property {string} AccountReference - Along with the business name, this value is also displayed to the customer in the STK pin prompt message. Maximum of 12 characters
+ * @property {string} TransactionDesc - Any additional information/comment that can be sent along with the request from you system.
+ */
+
+
+/**
+ * Initiate a STK pin prompt to a Safaricom mobile number.
+ * @function
+ * @public
+ * @param {STKPushObject} data 
+ * @param {RequestOptions} options 
+ * @example <caption>Initiate STK Push</caption>
+ * stkPush({
+    "Amount": 1,
+    "PartyA": 254708374149,
+    "PartyB": 174379,
+    "PhoneNumber": 254708374149,
+    "CallBackURL": "https://mydomain.com/path",
+    "AccountReference": "CompanyXLTD",
+    "TransactionDesc": "Payment of X"
+});
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function stkPush(data, options = {}) {
+  return async state => {
+
+    const { short_code, pass_key } = state.configuration;
+
+    const timestamp = util.getTimestamp()
+
+    const password = short_code + pass_key + timestamp;
+
+    const jsonBody = JSON.stringify({
+      ...data,
+      BusinessShortCode: short_code,
+      Password: Buffer.from(password).toString('base64'),
+      Timestamp: timestamp,
+      TransactionType: 'CustomerPayBillOnline',
+    });
+
+    const response = await util.request(state, 'POST', '/mpesa/stkpush/v1/processrequest', { ...options, body: jsonBody })
```

Comment:
We might need to  log a message here to help the users have a view of its success or current proceedings like:
`Initialised an STK Push`?